src/test/java/io/percy/selenium/TestServer.java

```diff
@@ -0,0 +1,98 @@
+package io.percy.selenium;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import com.sun.net.httpserver.HttpContext;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpServer;
+
+/**
+ * HTTP server that serves the static files that make our test app.
+ */
+class TestServer {
+    // Server port.
+    private static final Integer PORT = 8000;
+
+    // Location for static files in our test app.
+    private static final String TESTAPP_DIR = "src/test/resources/testapp/";
+    private static final String INDEX_FILE = "index.html";
+
+    // Recognized Mime type map (extension -> mimetype)
+    private static final Map<String, String> MIME_MAP = new HashMap<String, String>();
+    static {
+        MIME_MAP.put("html", "text/html");
+        MIME_MAP.put("js", "application/javascript");
+        MIME_MAP.put("css", "text/css");
+    }
+
+    private static ExecutorService executor;
+    private static HttpServer server;
+
+    public static HttpServer startServer() throws IOException {
+        executor = Executors.newFixedThreadPool(1);
+        server = HttpServer.create(new InetSocketAddress(PORT), 0);
+        HttpContext context = server.createContext("/");
+        context.setHandler(TestServer::handleRequest);
+        server.setExecutor(executor);
+        server.start();
+        return server;
+    }
+
+    public static void shutdown() {
+        if (server != null) {
+            server.stop(1);
+        }
+        if (executor != null) {
+            executor.shutdownNow();
+        }
+    }
+
+    private static void handleRequest(HttpExchange exchange) throws IOException {
+        String requestedPath = exchange.getRequestURI().getPath();
```

Comment:
Wow ok, I guess this is a thing! Thanks for making your way through this.

Reply:
FWIW, this is almost identical to what we have in the example app (just lightly simplified here). I could not find a convenient library to do programmatic simple static file serving over HTTP.